pkg/fission-cli/cmd/httptrigger/create.go

```diff
@@ -95,13 +104,10 @@ func (opts *CreateSubCommand) complete(input cli.Input) error {
 	}
 
 	prefix := input.String(flagkey.HtPrefix)
-	if len(prefix) != 0 && len(triggerUrl) > 2 {
-		console.Warn("Prefix will take precedence over URL/RelativeURL")
-	}
-
-	if len(prefix) == 0 && len(triggerUrl) < 2 {
-		console.Error("You need to supply either Prefix or URL/RelativeURL")
-		os.Exit(1)
+	if prefix == "/" {
+		return errors.New("url with only root path is not allowed")
```

Comment:
What will happen if we allow URL with root path prefix?

Reply:
It may give entire path space to function and by default, we also have internal routes.

Reply:
If we have a function hello which has endpoints `/home`, `/world` and user wants to expose it as `http://IP/home` / `http://IP/world`, then it would be possible if we kept the prefix as `/` . Internal route of the function would be `http://IP/Namespace/hello`